### PR TITLE
feat(Console): Introduce ConsoleMessage type

### LINKED ISSUE #744: Feature request: console level
Page offers a console event but it is not possible to know the level of the message (log, error...).

    page.on('console', function (...args) {
        // No way to know if this is an error
    })

Would it be possible to implement a way to know the level? I don't know what would be the cleanest way, either one more event per level:

    page.on('consoleError', ...)

or a new type of event:

    page.on('consoleWithType', function ({type, args}) ...

Basically here: https://github.com/GoogleChrome/puppeteer/blob/master/lib/Page.js#L308 the event type is totally ignored

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch introduces ConsoleMessage type and starts dispatching
it for the `'console'` event.

BREAKING CHANGE: this breaks the api of the `'console'` event.

Fixes #744.

### CODE REVIEW

File: lib/Page.js [hunk 1]
```diff
@@ -313,7 +313,10 @@ class Page extends EventEmitter {
       return;
     }
     const values = await Promise.all(event.args.map(arg => helper.serializeRemoteObject(this._client, arg)));
-    this.emit(Page.Events.Console, ...values);
+    const text = values.length && typeof values[0] === 'string' ? values[0] : null;
```
Comment (JoelEinbinder) [inline lib/Page.js]: Let's make text all the `values.join(" ")`

File: docs/api.md [hunk 2]
```diff
@@ -1093,6 +1097,21 @@ puppeteer.launch().then(async browser => {
 
 Dialog's type, could be one of the `alert`, `beforeunload`, `confirm` and `prompt`.
 
+### class: ConsoleMessage
```
Comment (JoelEinbinder) [inline docs/api.md:1100]: Inline this into the console event?

**Review by JoelEinbinder — APPROVED** [review]

File: docs/api.md (hunk 2 above)
Comment (aslushnikov) [reply docs/api.md:1100]: I think it's fine like this

**Comment by vsemozhetbyt** [comment]
It seems these fragments also need updating:

https://github.com/GoogleChrome/puppeteer/blame/0164b7cc4e60451665fafc2e8ccdb7855237ed29/docs/api.md#L517

https://github.com/GoogleChrome/puppeteer/blame/0164b7cc4e60451665fafc2e8ccdb7855237ed29/docs/api.md#L539

https://github.com/GoogleChrome/puppeteer/blame/ccf8fb168166ee12f42586e608d3557ff48edc27/README.md#L156

**Pull request merged**